wireguard-go/Wireguard-Go-FIPS-wolfCrypt-port.patch

```diff
@@ -1469,6 +1469,22 @@ index d81dae3..c866804 100644
  			peer.handshake.mutex.RUnlock()
  			sendf("protocol_version=1")
  			peer.endpoint.Lock()
--- 
-2.39.3 (Apple Git-146)
+diff --git a/device/noise-helpers.go b/device/noise-helpers.go
+index 6f2ed8c..a8934bb 100644
+--- a/device/noise-helpers.go
++++ b/device/noise-helpers.go
+@@ -89,12 +89,14 @@ func newPrivateKey() (sk NoisePrivateKey, err error) {
+             return sk, errors.New("Failed to make ECC key")
+         }
 
++        wolfSSL.PRIVATE_KEY_UNLOCK()
+         skLen := len(sk[:])
+         if ret := wolfSSL.Wc_ecc_export_private_only(&key, sk[:], &skLen); ret != 0 {
+             wolfSSL.Wc_FreeRng(&rng)
+             wolfSSL.Wc_ecc_free(&key)
+             return sk, errors.New("Failed to export private ECC key")
```

Comment:
just noticed this here, could it return out without calling wolfSSL.PRIVATE_KEY_LOCKS()?

Reply:
Very true. Fixed